### PR TITLE
(WIP) Update pause points

### LINKED ISSUE #5764: [PausePoints] iterate on the datastructure
PausePoints' current datastructure could use some tweaking

- it is a bit verbose 
- it is not optimized for lookups
- stepOver could be step

### Current 

```js
[
  {
    location: {line: 1, column: 3},
    types: {
      breakpoint: true,
      stepOver: false
    }
  }
]
```

### New

- Similar to the [BreakpointActorMap](https://searchfox.org/mozilla-central/source/devtools/server/actors/utils/breakpoint-actor-map.js#75-81)
- Keys the pause points by line & column to make lookups constant time
- a bit more terse than before.

```js
{
  1: {
    3: { breakpoint: true, step: true }
  }
}
```

### PR DESCRIPTION
**Pull request by jasonLaster**
Fixes  #5764 

### Summary of Changes

* switches to dictionary
* changes stepOver to step
* will add additional empty pause points

### CODE REVIEW

**Comment by darkwing** [comment]
Seeing test timeouts; kicking Travis to see if it's just an intermittent

**Comment by jasonLaster** [comment]
nah - it's me :)

**Comment by jasonLaster** [comment]
This is blocked on https://bugzilla.mozilla.org/show_bug.cgi?id=1450323 landing

File: src/utils/ast.js [hunk 1]
```diff
@@ -42,21 +43,24 @@ export function findBestMatchExpression(
...
   const breakpointLines = breakpoints.map(point => point.location.line);
 
   if (!selectedSource.text) {
     return [];
   }
+
   const lineCount = selectedSource.text.split("\n").length;
   const sourceLines = range(1, lineCount + 1);
-  return without(sourceLines, ...breakpointLines);
+  return xor(sourceLines, breakpointLines);
```
Comment (AnshulMalik) [inline src/utils/ast.js]: Nice!

Comment (jasonLaster) [reply src/utils/ast.js]: faster!

**Pull request merged**